test/extended/build_context_dir_test.go

```diff
@@ -0,0 +1,68 @@
+package extended
+
+import (
+	"testing"
+	"time"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/test/util"
+)
+
+func init() {
+	util.RequireDocker()
+	util.RequireServer()
+}
+
+func TestBuildContextDir(t *testing.T) {
+	build := mockBuildWithContextDir()
+	client := util.GetClient()
+	if _, err := client.Builds("contextdir").Create(build); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	retries := 0
+
+	for {
+		time.Sleep(time.Duration(1 * time.Second))
```

Comment:
Yes, there was a TODO before and I can't remember what led me to hide it :-) I will rework this to use watch.